front/change.form.php

```diff
@@ -139,16 +139,29 @@
    Html::header(Change::getTypeName(Session::getPluralNumber()), $_SERVER['PHP_SELF'], "helpdesk", "change");
    $change->display($_REQUEST);
 
-   if (isset($_GET['id']) && ($_GET['id'] > 0) && isset($_GET['_sol_to_kb'])) {
-      Ajax::createIframeModalWindow(
-         'savetokb',
-         KnowbaseItem::getFormURL() . '?_in_modal=1&item_itemtype=Change&item_items_id=' . $_GET['id'],
-         [
-            'title'         => __('Save solution to the knowledge base'),
-            'reloadonclose' => false,
-         ]
-      );
-      echo Html::scriptBlock('$(function() {' . Html::jsGetElementbyID('savetokb') . '.dialog("open"); });');
+   if (isset($_GET['id']) && ($_GET['id'] > 0)) {
+      $handled = false;
+      $url = KnowbaseItem::getFormURL() . '?_in_modal=1&item_itemtype=Change&item_items_id=' . $_GET['id'];
+      if (isset($_GET['_sol_to_kb'])) {
+         $handled = true;
+      } else if (isset($_GET['_fw_to_kb'])) {
+         $url .= '&fw_id=' . $_GET['_fw_to_kb'];
+         $handled = true;
+      } else if (isset($_GET['_task_to_kb'])) {
+         $url .= '&task_id=' . $_GET['_task_to_kb'];
+         $handled = true;
+      }
+      if ($handled) {
+         Ajax::createIframeModalWindow(
+            'savetokb',
+            $url,
+            [
+               'title'         => __('Save solution to the knowledge base'),
```

Comment:
As far as I understand, there are 3 possible cases; title here is OK for one only